lib/netext/grpcext/conn.go

```diff
@@ -58,9 +61,22 @@ type clientConnCloser interface {
 
 // Conn is a gRPC client connection.
 type Conn struct {
-	raw clientConnCloser
+	addr   string
+	shares atomic.Uint64
+	raw    clientConnCloser
 }
 
+var (
+	// Is there any other way?
+
+	//nolint:golint,gochecknoglobals
+	connections = &sync.Map{}
```

Comment:
To raise awareness on this, the `sync.Map` documentation mentions:

> The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys.

Does the use of it we make here fit those use cases? Otherwise I would suggest a good old map with explicit locking instead 👍🏻